server/db/043-add-number-of-alerts-to-sessions.sql

```diff
@@ -0,0 +1,37 @@
+DO $migration$
+    DECLARE migrationId INT;
+    DECLARE lastSuccessfulMigrationId INT;
+BEGIN
+    -- The migration ID of this file
+    migrationId := 43;
+
+    -- Get the migration ID of the last file to be successfully run
+    SELECT MAX(id) INTO lastSuccessfulMigrationId
+    FROM migrations;
+
+    -- Only execute this script if its migration ID is next after the last successful migration ID
+    IF migrationId - lastSuccessfulMigrationId = 1 THEN
+        -- ADD SCRIPT HERE
+        ALTER TABLE sessions
+        ADD COLUMN IF NOT EXISTS number_of_alerts INT;
+
+	-- Assume only 1 alert was sent for all sessions before this column was added
+	UPDATE sessions
+	SET number_of_alerts = 1
+	WHERE number_of_alerts IS NULL;
+
+	-- Don't allow new sessions to be created without setting number_of_alerts
+	ALTER TABLE sessions
+	ALTER COLUMN number_of_alerts
+	SET NOT NULL;
+
+	-- Add 'RESET' as a chatbot state
+	ALTER TYPE chatbot_state_enum
+	ADD VALUE IF NOT EXISTS 'RESET'
+	BEFORE 'COMPLETED';
```

Comment:
Just curious, does it matter which order these values are in the `enum`?

Reply:
Nope, it doesn't. I had a thought that the order of states in the enum followed the progression of the chatbot, and `COMPLETED` was "more completed" than `RESET`. But, on second thought, maybe it's more valuable to have `RESET` *after* `COMPLETED` maintain the integer equivalence of the other states?

Reply:
Thanks for clarifying. I'm happy either way, I was just curious!